### PR TITLE
dm: fix upstream switch job test

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: close #4287 

### What is changed and how it works?
- add dependency
- fix env variable
- add feishu notification

#### Tests 

 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: dm/tests/upstream_switch/case.sh [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 set -eu
 
+export DM_MASTER_EXTRA_ARG=""
```
Comment (lance6716) [inline dm/tests/upstream_switch/case.sh:5]: Will it overwrite the DM_MASTER_EXTRA_ARG set by GitHub secret?

**Review by lance6716 — APPROVED** [review]

File: dm/tests/upstream_switch/case.sh (hunk 1 above)
Comment (GMHDBJD) [reply dm/tests/upstream_switch/case.sh:5]: just same with other test https://github.com/pingcap/tiflow/blob/68c22f32b2090e83e5ae64ae3b1f25f1a9575ff8/dm/tests/run.sh#L6

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**